src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.isValidTriangle()) {
+      throw new Error('You cannot make a triangle with such sides');
+    }
+  }
+
+  isValidTriangle(): boolean {
+    return (
+      this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b);
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
 
+    const area = Math.sqrt(
+      halfPerimeter * (halfPerimeter - this.a)
+      * (halfPerimeter - this.b) * (halfPerimeter - this.c),
+    );
+
+    return Number(area.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+
+    if (this.radius <= 0) {
+      throw new Error('You cannot make a circle with such radius');
+    }
+  }
 
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
```

Comment:
maybe it would be better to write it like this: ```this.radius ** 2```

Reply:
It's a good point, let's fix it

Reply:
fixed